From 1d3f5743b6129cbca66638487a146e3f21f6c772 Mon Sep 17 00:00:00 2001
From: markw65 <mark_st@myosotissp.com>
Date: Sun, 8 May 2022 12:41:50 -0700
Subject: [PATCH] Fix monkey motion issues

This update was generated by finding all the devices in the manifest:
% devices=$(grep "iq:product " manifest.xml | sed -e 's/^.* id=//' -e 's/"\/.*$/"/' | jq -s)

And then pulling the device info for each device from Devices/*/compiler.json
% jq -r --argjson devices "$devices" -s 'map({key:.deviceId,value:{bitsPerPixel,size:(.deviceFamily|sub("^.*-";"")),partNumbers:(.partNumbers|map(.number))}})|from_entries as $info|$devices|sort_by($info[.].size)|map($info[.] as $i|$i.partNumbers|map("\(.)=Dancers_\($i.size)_\($i.bitsPerPixel).mm"))|flatten|.[]' ~/Library/Application\ Support/Garmin/ConnectIQ/Devices/*/compiler.json
---
 watchfaces/DanceDanceGarmin/asset/Dancers.mmm | 71 ++++++++++++++-----
 1 file changed, 54 insertions(+), 17 deletions(-)

diff --git a/watchfaces/DanceDanceGarmin/asset/Dancers.mmm b/watchfaces/DanceDanceGarmin/asset/Dancers.mmm
index bdd4d85..061aa97 100644
--- a/watchfaces/DanceDanceGarmin/asset/Dancers.mmm
+++ b/watchfaces/DanceDanceGarmin/asset/Dancers.mmm
@@ -4,38 +4,75 @@
 # Animation resource id: dancers
 # Video file: C:\Users\kralnic\eclipse-workspace\DanceDanceGarmin\asset\Dancers.yuv
 
-# 260x260 part number mappings
-006-B3499-00=Dancers_260x260_8.mm
-006-B3501-00=Dancers_260x260_8.mm
-006-B3289-00=Dancers_260x260_8.mm
-006-B3290-00=Dancers_260x260_8.mm
-006-B3225-00=Dancers_260x260_8.mm
-# 390x390 part number mappings
-006-B3226-00=Dancers_390x390_16.mm
-# 218x218 part number mappings
 006-B3500-00=Dancers_218x218_8.mm
+006-B3535-00=Dancers_218x218_8.mm
 006-B3498-00=Dancers_218x218_8.mm
+006-B3537-00=Dancers_218x218_8.mm
 006-B3224-00=Dancers_218x218_8.mm
-# 280x280 part number mappings
-006-B3291-00=Dancers_280x280_8.mm
-# 240x240 part number mappings
-006-B3076-00=Dancers_240x240_8.mm
-006-B3077-00=Dancers_240x240_8.mm
-006-B2886-00=Dancers_240x240_8.mm
-006-B2888-00=Dancers_240x240_8.mm
-006-B3113-00=Dancers_240x240_8.mm
+006-B3387-00=Dancers_218x218_8.mm
 006-B3110-00=Dancers_240x240_8.mm
+006-B3089-00=Dancers_240x240_8.mm
 006-B2900-00=Dancers_240x240_8.mm
+006-B3134-00=Dancers_240x240_8.mm
 006-B3111-00=Dancers_240x240_8.mm
+006-B3135-00=Dancers_240x240_8.mm
 006-B3287-00=Dancers_240x240_8.mm
+006-B3512-00=Dancers_240x240_8.mm
+006-B3764-00=Dancers_240x240_8.mm
+006-B3768-00=Dancers_240x240_8.mm
 006-B3288-00=Dancers_240x240_8.mm
+006-B3513-00=Dancers_240x240_8.mm
+006-B3765-00=Dancers_240x240_8.mm
+006-B3769-00=Dancers_240x240_8.mm
+006-B3076-00=Dancers_240x240_8.mm
+006-B3145-00=Dancers_240x240_8.mm
+006-B3914-00=Dancers_240x240_8.mm
+006-B3077-00=Dancers_240x240_8.mm
+006-B3321-00=Dancers_240x240_8.mm
+006-B3913-00=Dancers_240x240_8.mm
+006-B2886-00=Dancers_240x240_8.mm
+006-B3003-00=Dancers_240x240_8.mm
+006-B2888-00=Dancers_240x240_8.mm
+006-B3004-00=Dancers_240x240_8.mm
+006-B3113-00=Dancers_240x240_8.mm
+006-B3441-00=Dancers_240x240_8.mm
 006-B3624-00=Dancers_240x240_8.mm
+006-B3648-00=Dancers_240x240_8.mm
 006-B3251-00=Dancers_240x240_8.mm
+006-B3451-00=Dancers_240x240_8.mm
 006-B3247-00=Dancers_240x240_8.mm
+006-B3421-00=Dancers_240x240_8.mm
 006-B3248-00=Dancers_240x240_8.mm
+006-B3448-00=Dancers_240x240_8.mm
 006-B3249-00=Dancers_240x240_8.mm
+006-B3449-00=Dancers_240x240_8.mm
 006-B3246-00=Dancers_240x240_8.mm
+006-B3420-00=Dancers_240x240_8.mm
 006-B3250-00=Dancers_240x240_8.mm
+006-B3450-00=Dancers_240x240_8.mm
 006-B2700-00=Dancers_240x240_8.mm
+006-B2976-00=Dancers_240x240_8.mm
+006-B3446-00=Dancers_240x240_8.mm
 006-B2988-00=Dancers_240x240_8.mm
+006-B3163-00=Dancers_240x240_8.mm
 006-B3066-00=Dancers_240x240_8.mm
+006-B3289-00=Dancers_260x260_8.mm
+006-B3514-00=Dancers_260x260_8.mm
+006-B3766-00=Dancers_260x260_8.mm
+006-B3770-00=Dancers_260x260_8.mm
+006-B3290-00=Dancers_260x260_8.mm
+006-B3515-00=Dancers_260x260_8.mm
+006-B3782-00=Dancers_260x260_8.mm
+006-B3767-00=Dancers_260x260_8.mm
+006-B3771-00=Dancers_260x260_8.mm
+006-B3501-00=Dancers_260x260_8.mm
+006-B3536-00=Dancers_260x260_8.mm
+006-B3499-00=Dancers_260x260_8.mm
+006-B3538-00=Dancers_260x260_8.mm
+006-B3225-00=Dancers_260x260_8.mm
+006-B3388-00=Dancers_260x260_8.mm
+006-B3291-00=Dancers_280x280_8.mm
+006-B3516-00=Dancers_280x280_8.mm
+006-B3783-00=Dancers_280x280_8.mm
+006-B3226-00=Dancers_390x390_16.mm
+006-B3389-00=Dancers_390x390_16.mm
-- 
2.34.1

